src/Microsoft.Maintainability.Analyzers/VisualBasic/BasicRemoveUnusedLocals.vb

```diff
@@ -11,7 +14,72 @@ Namespace Microsoft.Maintainability.Analyzers
+        Private Shared ReadOnly s_localizableDescription As LocalizableString = New LocalizableResourceString(NameOf(MicrosoftMaintainabilityAnalyzersResources.RemoveUnusedLocalsDescription), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, GetType(MicrosoftMaintainabilityAnalyzersResources))
+
+        Friend Shared Rule As DiagnosticDescriptor = New DiagnosticDescriptor(
+                                                        RuleId,
+                                                        s_localizableTitle,
+                                                        s_localizableMessage,
+                                                        DiagnosticCategory.Performance,
+                                                        DiagnosticSeverity.Warning,
+                                                        True,
+                                                        s_localizableDescription,
+                                                        "https://msdn.microsoft.com/en-us/library/ms182278.aspx",
+                                                        WellKnownDiagnosticTags.Telemetry)
+
+        Public Overrides ReadOnly Property SupportedDiagnostics As ImmutableArray(Of DiagnosticDescriptor)
+            Get
+                Return ImmutableArray.Create(Rule)
+            End Get
+        End Property
+
+        Public Overrides Sub Initialize(analysisContext As AnalysisContext)
+
+            analysisContext.RegisterOperationBlockStartAction(
+            Sub(operationBlockContext)
+                Dim containingMethod = TryCast(operationBlockContext.OwningSymbol, IMethodSymbol)
+
+                If containingMethod IsNot Nothing Then
+                    Dim mightBecomeUnusedLocals = New HashSet(Of ILocalSymbol)()
+
+                    operationBlockContext.RegisterOperationAction(
+                    Sub(operationContext)
+                        Dim variables = DirectCast(operationContext.Operation, IVariableDeclarationStatement).Variables
+
+                        For Each variable In variables
+                            Dim local = variable.Variable
+
+                            mightBecomeUnusedLocals.Add(local)
+                        Next
+                    End Sub, OperationKind.VariableDeclarationStatement)
+
+                    operationBlockContext.RegisterOperationAction(
+                    Sub(operationContext)
+                        Dim localReferenceExpression As ILocalReferenceExpression = DirectCast(operationContext.Operation, ILocalReferenceExpression)
+                        Dim syntax = localReferenceExpression.Syntax
+
+                        ' The writeonly references should be ignored
+                        If syntax.Parent.IsKind(SyntaxKind.SimpleAssignmentStatement) AndAlso
```

Comment:
@JohnHamby We should consider exposing a property on all reference expressions indicating whether it's a read or write. I presume the bound nodes already have this information.